paasta_tools/tron_tools.py

```diff
@@ -231,6 +233,16 @@ def get_spark_paasta_pool(self):
     def get_env(self):
         env = super().get_env()
         spark_env = {}
+        if "aws_credentials_yaml" in self.config_dict:
```

Comment:
Are you sure you don't want to gate this behind `executor: spark` ? (line 246)?

Reply:
Good question, because These AWS-related env variables are of boarder scope than running spark drivers? I actually would love to hear more opinions on this. @drmorr0 @qui Thoughts?

Reply:
Most services and batches handle loading credentials themselves, when they start up. If some spark jobs have been written assuming that the credentials are already loaded, we can add this option to make things work, but I don't think we need it for other things.

Reply:
Yea, I think it should be gated.